cottoncandy/interfaces.py

```diff
@@ -599,10 +617,17 @@ def upload_raw_array(self, object_name, array, gzip=True, acl=DEFAULT_ACL, **met
             zipdata.seek(0)
             filestream = zipdata
             data_nbytes = get_fileobject_size(filestream)
-        else:
+        elif hasattr(numcodecs, compression.lower()): # if the mentioned compression type is in numcodecs
```

Comment:
Seems this logic can be more transparently handled by using:
`numcodecs.get_codec(dict(id=compression.lower())`
I can work on this later this week.

Reply:
Addressed in latest push.